src/MeMoModel.py

```diff
@@ -88,17 +89,60 @@ def annotate(self, allow_missing_dbs: bool = False) -> None:
 
 
 
-    def match(self, model2: MeMoModel, keep1ToMany:bool = True, output_names: bool = False, output_dbs: bool = False, keepUnmatched: bool = False) -> pd.DataFrame:
-        """ compares the metabolites of two models and returns a data frame with additional information 
+    def match(self,
+            model2: MeMoModel,
+            threshold_DB:float = 0,
+            threshold_name:float = 0.6,
+            threshold_total:float = 0,
+            keepAllMatches:bool = True,
+            output_names: bool = False,
+            output_dbs: bool = False,
+            keepUnmatched: bool = False) -> pd.DataFrame:
+        """ compares the metabolites of two models and returns a data frame
+        with additional information 
+        threshold_DB, threshold_name, threshold_total: value between 0 and 1. Defines a threshold on which a results should be kept
+        keepAllMatches: boolean, if true keeps all matches which survive the different thresholds, if false only top hits are being kept.
         output_names: If true, output the names of the metabolites that led to match based on levenshtein
+        output_dbs: If true, output the names of the database entries which have been matched on
+        keepUnmatched: returns also unmatched metabolites
         """
-        res_inchi = self.matchOnInchi(model2, keep1ToMany = keep1ToMany)
-        res_db = self.matchOnDB(model2, output_dbs = output_dbs, keep1ToMany= keep1ToMany)
-        res_name = self.matchOnName(model2, output_names = output_names, keep1ToMany= keep1ToMany)
+        res_inchi = self.matchOnInchi(model2)
+        res_db = self.matchOnDB(model2,
+                threshold = threshold_DB,
+                output_dbs = output_dbs)
+        res_name = self.matchOnName(model2,
+                threshold = threshold_name,
+                output_names = output_names)
 
         res = res_inchi.merge(res_db, how = "outer", on = ["met_id1","met_id2"],suffixes=["_inchi","_db"])
         res = res.merge(res_name, how = "outer", on = ["met_id1","met_id2"],suffixes=["","_name"])
+        res = res.rename(columns = {"charge_diff":"charge_diff_name", "inchi_string":"inchi_string_name"})
 
+        # calculate overall score in the result and remove NA
+        res.loc[pd.isna(res["inchi_score"]),res.columns =="inchi_score"] = 0
+        res.loc[pd.isna(res["DB_score"]),res.columns =="DB_score"] = 0
+        res.loc[pd.isna(res["Name_score"]),res.columns =="Name_score"] = 0
+
+        ##### this is the place where we could change the weighting of the scoring ####
+        res["total_score"] = (res["inchi_score"]*3 + res["DB_score"]*2+res["Name_score"])/6
+        ###############################################################################
+
+        # remove pairs with score < threshold (default = 0)
+        res = res.loc[res["total_score"]>threshold_total]
+        
+        # if not keep1ToMany, keep only the maximum score pair for each metabolite of both models
```

Comment:
a very very minor thing for code readibility, keep1ToMany should be renamed to keepAllMatches in the comment